batch/batch/server/database.py

```diff
@@ -119,7 +130,19 @@ def __init__(self, db):
         return await super().get_all_records()
 
     async def get_records(self, ids, fields=None):
-        return await super().get_record({'id': ids}, fields)
+        return await super().get_records({'id': ids}, fields)
+
+    async def get_records_where(self, condition):
+        return await super().get_records(condition)
 
     async def has_record(self, id):
         return await super().has_record({'id': id})
+
+    async def purge_finished_deleted_records(self):
+        async with self._db.pool.acquire() as conn:
+            async with conn.cursor() as cursor:
+                sql = f"DELETE FROM `{self.name}` WHERE `deleted` = TRUE AND `n_completed` = `n_jobs`"
```

Comment:
Could this batch still be open?  What's the relationship between deleted and open?

Reply:
I see below, delete closes.